pkg/controller/cluster_pools_pull_secret_provider/cluster_pools_pull_secret_provider_test.go

```diff
@@ -0,0 +1,241 @@
+package cluster_pools_pull_secret_provider
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/sirupsen/logrus"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/testhelper"
+)
+
+func init() {
+	if err := hivev1.AddToScheme(scheme.Scheme); err != nil {
+		panic(fmt.Sprintf("failed to register imagev1 scheme: %v", err))
+	}
+}
+
+var (
+	pool = &hivev1.ClusterPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ns",
+			Name:      "pool",
+		},
+		Spec: hivev1.ClusterPoolSpec{
+			PullSecretRef: &corev1.LocalObjectReference{
+				Name: "pull-secret",
+			},
+		},
+	}
+
+	anotherPool = &hivev1.ClusterPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ns",
+			Name:      "another-pool",
+		},
+		Spec: hivev1.ClusterPoolSpec{
+			PullSecretRef: &corev1.LocalObjectReference{
+				Name: "pull-secret",
+			},
+		},
+	}
+
+	anotherPoolInAnotherNS = &hivev1.ClusterPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "another-ns",
+			Name:      "another-pool",
+		},
+		Spec: hivev1.ClusterPoolSpec{
+			PullSecretRef: &corev1.LocalObjectReference{
+				Name: "pull-secret",
+			},
+		},
+	}
+
+	pool_without_pull_secret = &hivev1.ClusterPool{
```

Comment:
Nit: Please stick to the canonical camelCase

Reply:
Why did I keep doing this? Stupid me.

Reply:
Updated.